### ISSUE TITLE
disasm-a3xx.c : unknown(4,9)

### ISSUE THREAD

**Issue by metora** [freedreno-zz/freedreno#10]
Hello

I try to disassembly shader on Adreno 530, then it show [unknown(4,9)].
And I found [rcp rsq rsqh exp2h log2h sinpb2] in libllvm-qcom.so.
So I guess it has new instructions on Adreno 530.

**Comment by robclark** [comment]
yes, does look like a new cat4 instruction..  I guess if you have a simple shader that triggers the instruction, and the rest of the disassembled shader, it shouldn't be too hard to figure out what it is.

**Comment by metora** [comment]
The result I tried. 

OPC_RSQH            = _OPC(4, 9),
OPC_LOG2H           = _OPC(4, 10),
OPC_EXP2H           = _OPC(4, 11),

but I don't know what is sinpb2?

**Comment by robclark** [comment]
rsqh/log2h/exp2h... wonder if those are for half-precision (mediump)?  What happens if you try to use highp?

sinpb2, not really sure.. I'd have to see how it is used (ie. asm dump vs glsl)
